ext/random/csprng.c

```diff
@@ -0,0 +1,252 @@
+/*
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Tim Düsterhus <timwolla@php.net>                            |
+   |          Go Kudo <zeriyoshi@php.net>                                 |
+   +----------------------------------------------------------------------+
+*/
+
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "php.h"
+
+#include "Zend/zend_exceptions.h"
+
+#include "php_random.h"
+
+#if HAVE_UNISTD_H
+# include <unistd.h>
+#endif
+
+#ifdef PHP_WIN32
+# include "win32/time.h"
+# include "win32/winutil.h"
+# include <process.h>
+#endif
+
+#ifdef __linux__
+# include <sys/syscall.h>
+#endif
+
+#if HAVE_SYS_PARAM_H
+# include <sys/param.h>
+# if (__FreeBSD__ && __FreeBSD_version > 1200000) || (__DragonFly__ && __DragonFly_version >= 500700) || \
+     defined(__sun) || (defined(__NetBSD__) && __NetBSD_Version__ >= 1000000000)
+#  include <sys/random.h>
+# endif
+#endif
+
+#if HAVE_COMMONCRYPTO_COMMONRANDOM_H
+# include <CommonCrypto/CommonCryptoError.h>
+# include <CommonCrypto/CommonRandom.h>
+#endif
+
+#if __has_feature(memory_sanitizer)
+# include <sanitizer/msan_interface.h>
+#endif
+
+PHPAPI int php_random_bytes(void *bytes, size_t size, bool should_throw)
```

Comment:
```suggestion
PHPAPI zend_result php_random_bytes(void *bytes, size_t size, bool should_throw)
```

Reply:
I've noticed that myself, having followed the discussion on the ML. However I've intentionally *not included* this change in this PR to make it immediately obvious that it's a simple moving around when diffing the files. I consider this important for "trustworthiness" / to keep a clear "audit log".

Will make the change in a dedicated follow-up commit / PR.